echobot/echobot.c

```diff
@@ -40,9 +40,13 @@ Tox *create_tox()
 
         uint8_t *savedata = malloc(fsize);
```

Comment:
Since you're adding error checks, should this malloc also be guarded?

Reply:
Done.